Config file not found on Windows if project path contains spaces

-   Laravel Mix Version: laravel-mix@6.0.0-beta.15
-   Node Version v14.15.1
-   NPM Version 6.14.9
-   OS: Windows 10 Pro (Version 2004)

### Description:
Issue #2621 does not address the problem on Windows. While quoting is technically the correct solution, having node.js spawn a process with parameters containing spaces [will not pass them correctly](https://github.com/nodejs/node-v0.x-archive/issues/25895) to the underlying process as quotes are being removed.
While someone came up with a [quirky workaround](https://github.com/nodejs/node-v0.x-archive/issues/25895#issuecomment-180402332), I think it would likely be preferable to have [the webpack config file](https://github.com/JeffreyWay/laravel-mix/blob/b2777fc8530608d95d6813009a7e7607e27b9b47/bin/cli.js#L60) be passed with a relative path instead if applicable by defining `let config = require.resolve('../setup/webpack.config.js');` and then passing the config as such: `"${config.substr(config.startsWith(process.cwd()) ? process.cwd().length + 1 : 0)}"`

It's certainly not a perfect solution, but it would address the most common error scenario on Windows, while not causing OS-specific bloat where it doesn't belong.

### Steps To Reproduce:
Run `mix` on Windows while having the project reside in a path containing one or more spaces.

Sorry, but I tried out both the latest npm version (supposedly 6.0.5) and the current master branch. Both of which still observe the same error when using a setup with standard installation paths. What change is supposed to address this problem?

Failure to launch:
```
"C:\Program Files\nodejs\node.exe" "C:\Program Files\nodejs\node_modules\npm\bin\npm-cli.js" run watch-poll --scripts-prepend-node-path=auto

> @ watch-poll C:\Program Files\Apache\htdocs\xxxxx
> mix watch -- --watch-options-poll=1000

[webpack-cli] Running multiple commands at the same time is not possible
[webpack-cli] Found commands: 'bundle', 'Files\Apache\htdocs\xxxxx\node_modules\laravel-mix\setup\webpack.config.js'
[webpack-cli] Run 'webpack --help' to see available commands and options
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! @ watch-poll: `mix watch -- --watch-options-poll=1000`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the @ watch-poll script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\---\AppData\Roaming\npm-cache\_logs\2020-12-28T21_07_36_665Z-debug.log

Process finished with exit code 2
```

For completeness the `2020-12-28T21_07_36_665Z-debug.log`:
```
0 info it worked if it ends with ok
1 verbose cli [
1 verbose cli   'C:\\Program Files\\nodejs\\node.exe',
1 verbose cli   'C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'run',
1 verbose cli   'watch-poll',
1 verbose cli   '--scripts-prepend-node-path=auto'
1 verbose cli ]
2 info using npm@6.14.10
3 info using node@v14.15.1
4 verbose run-script [ 'prewatch-poll', 'watch-poll', 'postwatch-poll' ]
5 info lifecycle @~prewatch-poll: @
6 info lifecycle @~watch-poll: @
7 verbose lifecycle @~watch-poll: unsafe-perm in lifecycle true
8 verbose lifecycle @~watch-poll: PATH: C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin;C:\Program Files\Apache\htdocs\xxxxx\node_modules\.bin;C:\Python27\;C:\Python27\Scripts;C:\Program Files\ImageMagick-7.0.8-Q16;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\IDM Computer Solutions\UltraEdit;C:\ProgramData\ComposerSetup\bin;C:\Program Files\AMD\AMDuProf\bin;C:\Program Files\Calibre2\;C:\Program Files\MiKTeX 2.9\miktex\bin\x64\;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\TortoiseGit\bin;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\Geth;C:\Program Files (x86)\Gpg4win\..\GnuPG\bin;C:\Program Files\Inkscape\bin;C:\Program Files\Apache\php74;C:\Program Files\nodejs\;C:\Program Files\Git\cmd;C:\Users\---\AppData\Local\JSONBuddy;C:\Users\---\AppData\Local\Microsoft\WindowsApps;C:\Users\---\AppData\Roaming\Composer\vendor\bin;C:\Users\---\AppData\Roaming\npm
9 verbose lifecycle @~watch-poll: CWD: C:\Program Files\Apache\htdocs\xxxxx
10 silly lifecycle @~watch-poll: Args: [ '/d /s /c', 'mix watch -- --watch-options-poll=1000' ]
11 silly lifecycle @~watch-poll: Returned: code: 2  signal: null
12 info lifecycle @~watch-poll: Failed to exec watch-poll script
13 verbose stack Error: @ watch-poll: `mix watch -- --watch-options-poll=1000`
13 verbose stack Exit status 2
13 verbose stack     at EventEmitter.<anonymous> (C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:315:20)
13 verbose stack     at ChildProcess.<anonymous> (C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\lib\spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:315:20)
13 verbose stack     at maybeClose (internal/child_process.js:1048:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5)
14 verbose pkgid @
15 verbose cwd C:\Program Files\Apache\htdocs\xxxxx
16 verbose Windows_NT 10.0.19041
17 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "watch-poll" "--scripts-prepend-node-path=auto"
18 verbose node v14.15.1
19 verbose npm  v6.14.10
20 error code ELIFECYCLE
21 error errno 2
22 error @ watch-poll: `mix watch -- --watch-options-poll=1000`
22 error Exit status 2
23 error Failed at the @ watch-poll script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 2, true ]
```

Unfortunately, it does not work.
It seems now.......
$ npm run prod

> @ prod D:\Current Projects\practice-laravel
> npm run production


> @ production D:\Current Projects\practice-laravel
> cross-env NODE_ENV=production node_modules/webpack/bin/webpack.js --no-progres
s --config=node_modules/laravel-mix/setup/webpack.config.js

error: unknown option '--no-progress'
[webpack-cli] Run 'webpack --help' to see available commands and options
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! @ production: `cross-env NODE_ENV=production node_modules/webpack/bin/w
ebpack.js --no-progress --config=node_modules/laravel-mix/setup/webpack.config.j
s`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the @ production script.
npm ERR! This is probably not a problem with npm. There is likely additional log
ging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\ASUS\AppData\Roaming\npm-cache\_logs\2021-01-08T14_51_12_2
11Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! @ prod: `npm run production`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the @ prod script.
npm ERR! This is probably not a problem with npm. There is likely additional log
ging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\ASUS\AppData\Roaming\npm-cache\_logs\2021-01-08T14_51_12_4
81Z-debug.log

--- a/bin/cli.js
+++ b/bin/cli.js
@@ -2,6 +2,7 @@
 
 const { Command } = require('commander');
 const { spawn } = require('child_process');
+const path = require('path');
 
 run();
 
@@ -46,18 +47,25 @@ async function run() {
  * @param {string[]} args
  */
 async function executeScript(cmd, opts, args = []) {
-    const env =
-        opts.production
-            ? 'production'
-            : (isTesting() && process.env.NODE_ENV === 'test') || !process.env.NODE_ENV
-                ? 'development'
-                : process.env.NODE_ENV;
+    const env = opts.production
+        ? 'production'
+        : (isTesting() && process.env.NODE_ENV === 'test') || !process.env.NODE_ENV
+        ? 'development'
+        : process.env.NODE_ENV;
+
+    // We MUST use a relative path because the files
+    // created by npm dont correctly handle paths
+    // containg spaces on Windows (yarn does)
+    const configPath = path.relative(
+        process.cwd(),
+        require.resolve('../setup/webpack.config.js')
+    );
 
     let script = [
         `cross-env NODE_ENV=${env}`,
         `MIX_FILE="${opts.mixConfig}"`,
         commandScript(cmd, opts),
-        `--config="${require.resolve('../setup/webpack.config.js')}"`,
+        `--config="${configPath}"`,
         ...quoteArgs(args)
     ].join(' ');
 

--- a/test/unit/cli.js
+++ b/test/unit/cli.js
@@ -1,14 +1,25 @@
 import test from 'ava';
+import path from 'path';
 import { cli } from '../helpers/cli';
 
 const mix = cli({ testing: true });
+
+/** @type {string} */
+let configPath = '';
+
+test.before(() => {
+    configPath = path.relative(
+        process.cwd(),
+        require.resolve('../../setup/webpack.config.js')
+    );
+});
 
 test('it calls webpack in development mode', async t => {
     let { stdout } = await mix();
 
     t.is(
         'cross-env NODE_ENV=development MIX_FILE="webpack.mix" npx webpack --progress --config="' +
-            require.resolve('../../setup/webpack.config.js') +
+            configPath +
             '"',
         stdout
     );
@@ -19,7 +30,7 @@ test('it calls webpack in production mode', async t => {
 
     t.is(
         'cross-env NODE_ENV=production MIX_FILE="webpack.mix" npx webpack --progress --config="' +
-            require.resolve('../../setup/webpack.config.js') +
+            configPath +
             '"',
         stdout
     );
@@ -30,7 +41,7 @@ test('it calls webpack with watch mode', async t => {
 
     t.is(
         'cross-env NODE_ENV=development MIX_FILE="webpack.mix" npx webpack --progress --watch --config="' +
-            require.resolve('../../setup/webpack.config.js') +
+            configPath +
             '"',
         stdout
     );
@@ -41,7 +52,7 @@ test('it calls webpack with watch mode using polling', async t => {
 
     t.is(
         'cross-env NODE_ENV=development MIX_FILE="webpack.mix" npx webpack --progress --watch --config="' +
-            require.resolve('../../setup/webpack.config.js') +
+            configPath +
             '"' +
             ' --watch-poll',
         stdout
@@ -53,7 +64,7 @@ test('it calls webpack with hot reloading', async t => {
 
     t.is(
         'cross-env NODE_ENV=development MIX_FILE="webpack.mix" npx webpack serve --hot --config="' +
-            require.resolve('../../setup/webpack.config.js') +
+            configPath +
             '"',
         stdout
     );
@@ -64,7 +75,7 @@ test('it calls webpack with hot reloading using polling', async t => {
 
     t.is(
         'cross-env NODE_ENV=development MIX_FILE="webpack.mix" npx webpack serve --hot --config="' +
-            require.resolve('../../setup/webpack.config.js') +
+            configPath +
             '"' +
             ' --watch-poll',
         stdout
@@ -76,7 +87,7 @@ test('it calls webpack with quoted key value pair command arguments', async t =>
 
     t.is(
         'cross-env NODE_ENV=development MIX_FILE="webpack.mix" npx webpack --progress --config="' +
-            require.resolve('../../setup/webpack.config.js') +
+            configPath +
             '"' +
             ' --env foo="bar baz" foo="bar=baz"',
         stdout
@@ -94,7 +105,7 @@ test('it calls webpack with custom node_env', async t => {
 
     t.is(
         'cross-env NODE_ENV=foobar MIX_FILE="webpack.mix" npx webpack --progress --config="' +
-            require.resolve('../../setup/webpack.config.js') +
+            configPath +
             '"',
         stdout
     );
@@ -107,7 +118,7 @@ test('it disables progress reporting when not using a terminal', async t => {
 
     t.is(
         'cross-env NODE_ENV=development MIX_FILE="webpack.mix" npx webpack --config="' +
-            require.resolve('../../setup/webpack.config.js') +
+            configPath +
             '"',
         stdout
     );

